MediaGallery/MediaGallery/SaveMedia.android.cs

```diff
@@ -20,23 +21,65 @@ namespace NativeMedia
 {
     public static partial class MediaGallery
     {
-        static async Task PlatformSaveAsync(MediaFileType type, byte[] data, string fileName)
+        public static async Task SaveAsync(Activity activity, MediaFileType type, byte[] data, string fileName)
+        {
+            await CheckPossibilitySave();
+            if (!(data?.Length > 0))
+                throw new ArgumentNullException(nameof(data));
+            CheckFileName(fileName);
+            if(activity == null)
+                throw new ArgumentNullException(nameof(activity));
+            
+            await PlatformSaveAsync(activity, type, data, fileName);
+        }
+
+        public static async Task SaveAsync(Activity activity, MediaFileType type, string filePath)
+        {
+            await CheckPossibilitySave();
+            if (string.IsNullOrWhiteSpace(filePath) || !System.IO.File.Exists(filePath))
+                throw new ArgumentException(nameof(filePath));
+            if(activity == null)
+                throw new ArgumentNullException(nameof(activity));
+            
+            await PlatformSaveAsync(activity, type, filePath);
+        }
+
+        public static async Task SaveAsync(Activity activity, MediaFileType type, Stream fileStream, string fileName)
+        {
+            await CheckPossibilitySave();
+            if (fileStream == null)
+                throw new ArgumentNullException(nameof(fileStream));
+            CheckFileName(fileName);
+            if(activity == null)
+                throw new ArgumentNullException(nameof(activity));
```

Comment:
I saw this null check in a lot of places, you can create a helper method to check it for you